index.js

```diff
@@ -8,41 +7,31 @@ class CancelError extends Error {
 	}
 }
 
-const generate = willResolve => function (ms, value) {
-	ms = ms || 0;
-	const useValue = (arguments.length > 1);
-	let result = value;
-
-	const delaying = defer();
-	const promise = delaying.promise;
-
-	let timeout = setTimeout(() => {
-		const settle = willResolve ? delaying.resolve : delaying.reject;
-		settle(result);
-	}, ms);
-
-	const thunk = thunkResult => {
-		if (!useValue) {
-			result = thunkResult;
-		}
-		return promise;
-	};
-
-	thunk.then = promise.then.bind(promise);
-	thunk.catch = promise.catch.bind(promise);
-	thunk._actualPromise = promise;
-
-	thunk.cancel = () => {
-		if (timeout) {
-			clearTimeout(timeout);
-			timeout = null;
-			delaying.reject(new CancelError('Delay canceled'));
+const createDelay = willResolve => (ms, value) => {
+	let timeoutId = null;
+	let internalReject = null;
+
+	const delayPromise = new Promise((resolve, reject) => {
+		ms = ms || 0;
```

Comment:
We don't actually need this as `setTimeout` will default to `0`.

Reply:
Will do.  I figured this was being used to catch other falsey values which would cause undefined behavior.  Doesn't matter to me, because input validation would obviously be a better approach to that problem.